File: erezervisi_api/eRezervisi/eRezervisi.Common.Dtos/Search/SearchResponse.cs
using eRezervisi.Common.Dtos.AccommodationUnit;
using eRezervisi.Common.Dtos.User;

namespace eRezervisi.Common.Dtos.Search
{
    public class SearchResponse
    {
        public List<AccommodationUnitGetShortDto> AccommodationUnits { get; set; } = new();
        public List<UserGetShortDto> Users { get; set; } = new();
    }
}

File: erezervisi_mobile/erezervisi_mobile/lib/models/responses/search/search_response.dart
import 'package:erezervisi_mobile/models/responses/accommodation_unit/accommodation_unit_get_short_dto.dart';
import 'package:erezervisi_mobile/models/responses/user/user_get_short_dto.dart';

class SearchResponse {
  late List<AccommodationUnitGetShortDto> accommodationUnits;
  late List<UserGetShortDto> users;

  SearchResponse({required this.accommodationUnits, required this.users});

  factory SearchResponse.fromJson(Map<String, dynamic> json) {
    List<AccommodationUnitGetShortDto> accommodationUnits =
        (json['accommodationUnits'] as List<AccommodationUnitGetShortDto>)
            .map((item) => AccommodationUnitGetShortDto.fromJson(json))
            .toList();

    List<UserGetShortDto> users = (json['users'] as List<UserGetShortDto>)
        .map((item) => UserGetShortDto.fromJson(json))
        .toList();

    return SearchResponse(accommodationUnits: accommodationUnits, users: users);
  }
}
